client/app/scripts/controllers/repos.js

```diff
@@ -70,8 +70,9 @@ angular.module('clientApp')
             // Add build info to the repo info.
             var data = response.data.data[0];
             $scope.repos[key].build = {
-              enabled: true,
-              id: data.id
+              enabled: data.enabled,
+              id: data.id,
+              disable_reason: data.disable_reason
```

Comment:
hook_node_presave() - `throw \Excpetion`

You will now the disable reason by the exception message.